Content.Shared/Chat/TypingIndicator/TypingIndicatorComponent.cs

```diff
@@ -14,7 +15,13 @@ public sealed partial class TypingIndicatorComponent : Component
     /// <summary>
     ///     Prototype id that store all visual info about typing indicator.
     /// </summary>
-    [ViewVariables(VVAccess.ReadWrite)]
-    [DataField("proto", customTypeSerializer: typeof(PrototypeIdSerializer<TypingIndicatorPrototype>))]
-    public string Prototype = SharedTypingIndicatorSystem.InitialIndicatorId;
+    [DataField]
+    public ProtoId<TypingIndicatorPrototype> DefaultTypingIndicator = "default";
+
+    /// <summary>
+    ///     A list of all indicators that override the default one. E.g if you put on
+    ///     a moth mask the list would have both the lawyer and the moth indicator.
+    /// </summary>
+    [DataField]
+    public List<ProtoId<TypingIndicatorPrototype>> TypingIndicatorOverrideList = new List<ProtoId<TypingIndicatorPrototype>>();
 }
```

Comment:
This needs networking or it's just going to break.

Reply:
I honestly don't really know what that means! Is it just adding a tag thing or is it more complicated?

Reply:
It's more than just a tag, but it's not too bad. 

Alright. So. 

The problem is that due to the way that clients delete and reload objects that get far enough away, you're going to wind up with the list of indicators getting desynced between clients when people move around. The answer is to have the server provide the correct information when a client reloads another player. To do that, we need to set up networking on the typing indicator component and system. 

To do this, we first need to set up the component. Thankfully, it looks like the component is already in Shared and tagged with NetworkedComponent, so that saves us some work: that gets the component replicated into clients, but it doesn't handle copying the component's _state_ (basically the values of its fields). Thankfully again, this component is simple enough that sourcegen makes this easy. 

Add `[AutoGenerateComponentState]` to the component. That marks it for sourcegen to look at. You'll get an error message about not having any marked fields. Just add `[AutoNetworkedField]` to the fields that should be replicated (in this case, the prototype list).

That sets up the component to transfer state from server to clients. Now we just need to tell the engine _when_ we need the server to send that data.

In the typing indicator system, whenever the code modifies the component's state, it should flag it as dirty. This will tell the engine that the server's version of the component has changed and should be replicated to clients. So after changing the list's contents in the gotEquipped/gotUnequipped event handlers, call `Dirty(uid, component)`.

And that should do it!

Man, we should probably write up some more in-depth tutorials on this.

Reply:
OK that make a lot of sense! Now I get why you have to do the dirty command 😆

Also you should probably write it up because I can 100% guarantee I will come back to your comment every time I need to do networking stuff to make sure I'm doing it correctly. I'm sure other people would love to see this!

Thanks so much for the detailed explanation :)